core/dbt/parser/schemas.py

```diff
@@ -927,6 +929,28 @@ def _validate_constraint_prerequisites(self, model_node: ModelNode):
         if str(model_node.language) != "sql":
```

Comment:
I actually think that from [line 923](https://github.com/dbt-labs/dbt-core/pull/9700/files#diff-d6c689f513ec4e0f5ddbae6bdc92a08dfcb930d5fa8b2fa70c87fe9c366a53e7R923) -- this should be a separate method that is called regardless of whether `contract_config.enforced` is true or not. Currently, `_validate_constraint_prerequisites` is only called when the contract enforcement is enabled.

The warn_or_error issues above seem directly related to constraint validation, whereas the constraint parsing is independent and should be consistent on every project regardless of whether the contract is being validated or not.